### PR TITLE
Added capability to RasterImage layer to read geogeaphic info from .aux.xml file

### PR DESCRIPTION
**Pull request by ma15569**
See mail https://sourceforge.net/p/jump-pilot/mailman/message/37833644/ for more details

### CODE REVIEW

File: src/org/openjump/core/ccordsys/utils/ProjUtils.java [hunk 1]
```diff
@@ -486,6 +487,7 @@ public static SRSInfo getSRSInfoFromLayerSource(RasterImageLayer layer)
         // fallthrough
         if (srsInfo == null) {
             srsInfo = ProjUtils.getSRSInfoFromAuxiliaryFile(fileSourcePath);
+             srsInfo.setSource(AUXFILE_SRS);
```
Comment (edeso) [inline src/org/openjump/core/ccordsys/utils/ProjUtils.java:490]: this should be set in getSRSInfoFromAuxiliaryFile() **and only if** the ".aux.xml" file is indeed available and the srs is read from it.

File: src/org/openjump/core/rasterimage/AddRasterImageLayerWizard.java [hunk 2]
```diff
@@ -263,175 +263,114 @@ private void addImage(WorkbenchContext context, Envelope envelope,
...
-        worldFileHandler = new WorldFileHandler(fileName,
-                alwaysLookForTFWExtension);
-
-        if (imageDimensions == null) {
-            // logger.printError("can not determine image dimensions");
-            context.getWorkbench()
-                    .getFrame()
-                    .warnUser(
-                            I18N.getInstance().get("org.openjump.core.rasterimage.AddRasterImageLayerWizard.can-not-determine-image-dimensions"));
-            return null;
```
Comment (edeso) [inline src/org/openjump/core/rasterimage/AddRasterImageLayerWizard.java:282]: this is wrong. you added tabs to the indentation (spaces only before) and did some unneeded reformatting. i cannot see clearly what you changed here.

please fix the whole file. compare with previous version from branch main.

**Comment by jratike80** [comment]
I believe (not sure) that aux.xml files may exist even without georeferencing info. Does the code skip aux.xml in such case and search for world files etc.?

**Comment by edeso** [comment]
hey Jukka,

it's pretty convoluted by now. but yeah, in theory it should try all methods until one succeeds. should be this part. try geotiff or if not avail, try sidecar file.

https://github.com/openjump-gis/openjump/blob/main/src/org/openjump/core/ccordsys/utils/ProjUtils.java#L468-L499

File: src/org/openjump/core/ccordsys/utils/ProjUtils.java (hunk 1 above)
Comment (ma15569) [reply src/org/openjump/core/ccordsys/utils/ProjUtils.java:490]: I will remove this line as I didn't realize that ProjUtils.getSRSInfoFromAuxiliaryFile() has already a setSource that points to the path of aux file (strangely it fails sometimes, I will inverstigate later)

File: src/language/jump.properties [hunk 3]
```diff
@@ -2964,6 +2964,7 @@ ui.zoom.ZoomToCoordinatePlugIn.enter-two-values=Enter two values!
 ui.zoom.ZoomToCoordinatePlugIn.{0}-is-not-a-number=''{0}'' is not a number!
 org.openjump.core.ccordsys.No-srs=No SRS info recorded from data source
 org.openjump.core.ccordsys.Embedded-srs=SRS info is included in data source
+org.openjump.core.ccordsys.AuxFile-srs=SRS info is included into an auxiliary file
```
[the remark this answers is not in the document]
Comment (ma15569) [reply src/language/jump.properties]: This codes will be removed. See comment in [src/org/openjump/core/ccordsys/utils/ProjUtils.java](https://github.com/openjump-gis/openjump/pull/97/files/df818342b30c031c9c337605d8324c2c20eeb07a#diff-ba250e9f8be116c2fc02528c7b1b08e2b64faba706807d1d8f351bbe512a4573)

File: src/org/openjump/core/rasterimage/AddRasterImageLayerWizard.java (hunk 2 above)
Comment (ma15569) [reply src/org/openjump/core/rasterimage/AddRasterImageLayerWizard.java:282]: I see the tabs+spaces. I will correct it

Should I close this pull request and open a new one?

Comment (edeso) [reply src/org/openjump/core/rasterimage/AddRasterImageLayerWizard.java:282]: no. you can just push to the branch this PR is based on :)

**Comment by ma15569** [comment]
Hi Ede,
this request seems still in progress.

**Comment by ma15569** [comment]
😕

**Comment by edeso** [comment]
thx for the reminder. just ravelling.

will try to fit in a review while riding the train Mo or Tue :)

File: src/language/jump_es.properties [hunk 4]
```diff
@@ -3008,3 +3008,4 @@ org.openjump.core.ui.plugin.tools.MergeSelectedPolygonsWithNeighbourPlugIn.merge
 org.openjump.core.ui.plugin.tools.MergeSelectedPolygonsWithNeighbourPlugIn.minimum-area=#T:minimum area
 org.openjump.core.ui.plugin.tools.MergeSelectedPolygonsWithNeighbourPlugIn.Skip-null-values=#T:Skip features with null attribute value
 jump.features-processed=elementos procesados
+org.openjump.core.ccordsys.AuxFile-srs=SRS se incluye en un archivo auxiliar
```
Comment (edeso) [inline src/language/jump_es.properties:3011]: this new key needs to be added to **all language files**
especially to the english version `src/language/jump.properties` with english translation

File: src/org/openjump/core/rasterimage/AddRasterImageLayerWizard.java [hunk 5]
```diff
@@ -255,6 +260,12 @@ private void addImage(WorkbenchContext context, Envelope envelope,
      * first pixel and not the center. I have corrected this in
      * WorldFileHandler.readWorldFile()
      * 
+     * [Giuseppe Aruta - 04/27/2023]. The method has been "slimed" from the previous version:<br>
+     * <li>a) The first part that build the Envelope from image info has been removed and delegated
```
Comment (edeso) [inline src/org/openjump/core/rasterimage/AddRasterImageLayerWizard.java:264]: do you mean "slimmed down" meaning reduced in complexity/size?

**Review by edeso — CHANGES_REQUESTED** [review]
just two small issues, the files being the bigger one. getting there :)

**Comment by ma15569** [comment]
Hi Ede,

a) I removed the code
 _org.openjump.core.ccordsys.AuxFile-srs=SRS se incluye en un archivo auxiliar as_ 
from the Spanis lang file as it is not used anymore
Meanwhile I added new translations to that file

b) AdddRasterIageLayerWizard.  I correct the typo from "slimed"  to "optimized".

Not sure if the corrections are visible on main brench

Peppe

**Comment by ma15569** [comment]
??

**Comment by edeso** [comment]
this line in _ProjUtils_
```
private static final String AUXFILE_SRS = I18N.getInstance().get("org.openjump.core.ccordsys.AuxFile-srs");
```
is obsolete now, is it not?

**Comment by edeso** [comment]
is this still ongoing or was it resolved inbetween?

**Comment by ma15569** [comment]
Hi Ede, I prefer to abandon this request and focus on developing my own external plugins, where I have more possibilities to manage the code. This option has however been added to my Raster Tools. which have a parallel system of importing raster data. If you don't mind, I will close the request.

**Comment by edeso** [comment]
>  If you don't mind, I will close the request.

sure. go ahead :)

**Pull request closed without merging**